code/modules/language/_language.dm

```diff
@@ -0,0 +1,164 @@
+	var/default_name_syllable_min = 2
+	/// By default, random names picks this many syllables (max)
+	var/default_name_syllable_max = 4
+	/// What char to place in between randomly generated names
+	var/random_name_spacer = " "
+
+/// Checks whether we should display the language icon to the passed hearer.
+/datum/language/proc/display_icon(atom/movable/hearer)
+	var/understands = hearer.has_language(src.type)
+	if((flags & LANGUAGE_HIDE_ICON_IF_UNDERSTOOD) && understands)
+		return FALSE
+	if((flags & LANGUAGE_HIDE_ICON_IF_NOT_UNDERSTOOD) && !understands)
+		return FALSE
+	return TRUE
+
+/// Returns the icon to display in the chat window when speaking this language.
+/datum/language/proc/get_icon()
+	var/datum/asset/spritesheet/sheet = get_asset_datum(/datum/asset/spritesheet/chat)
+	return sheet.icon_tag("language-[icon_state]")
+
+/// Simple helper for getting a default firstname lastname
+/datum/language/proc/default_name(gender = NEUTER)
+	if(gender != MALE)
+		gender = pick(MALE, FEMALE)
+	if(gender == FEMALE)
+		return capitalize(pick(GLOB.first_names_female)) + " " + capitalize(pick(GLOB.last_names))
+	return capitalize(pick(GLOB.first_names_male)) + " " + capitalize(pick(GLOB.last_names))
+
+
+/**
+ * Generates a random name this language would use.
+ *
+ * * gender: What gender to generate from, if neuter / plural coin flips between male and female
+ * * name_count: How many names to generate in, by default 2, for firstname lastname
+ * * syllable_count: How many syllables to generate in each name, min
+ * * syllable_max: How many syllables to generate in each name, max
+ * * force_use_syllables: If the name should be generated from the syllables list.
+ * Only used for subtypes which implement custom name lists. Also requires the language has syllables set.
+ */
+/datum/language/proc/get_random_name(
+	gender = NEUTER,
+	name_count = default_name_count,
+	syllable_min = default_name_syllable_min,
+	syllable_max = default_name_syllable_max,
+	force_use_syllables = FALSE,
+)
+	if(gender != MALE)
+		gender = pick(MALE, FEMALE)
+	if(!length(syllables) || always_use_default_namelist)
+		return default_name(gender)
+
+	var/list/full_name = list()
+	for(var/i in 1 to name_count)
+		var/new_name = ""
+		for(var/j in 1 to rand(default_name_syllable_min, default_name_syllable_max))
+			new_name += pick_weight_recursive(syllables)
+		full_name += capitalize(LOWER_TEXT(new_name))
+
+	return jointext(full_name, random_name_spacer)
+
+/// Generates a random name, and attempts to ensure it is unique (IE, no other mob in the world has it)
+/datum/language/proc/get_random_unique_name(...)
+	var/result = get_random_name(arglist(args))
+	for(var/i in 1 to 10)
+		if(!findname(result))
+			break
+		result = get_random_name(arglist(args))
+
+	return result
+
+/datum/language/proc/check_cache(input)
+	var/lookup = scramble_cache[input]
+	if(lookup)
+		scramble_cache -= input
+		scramble_cache[input] = lookup
+	. = lookup
```

Comment:
```suggestion
	return lookup
```

unless there's a reason why you're doing it like this

Reply:
I didn't touch this code

Reply:
ah i think i just noticed it since it was being shuffled around. probably fine to leave it like this but it made me squint